### ISSUE TITLE
Benchmarks should operate on square whole images

### ISSUE THREAD

**Issue by DanHickstein** [PyAbel/PyAbel#207]
Currently, abel.benchmark.AbelTiming passes, for example, a `101x101` pixel *half-image* to each transform method. This corresponds to a `101x201` pixel *whole-image* when we apply the cylindrical (left-right) symmetry.

In most experimental situations, people are recording both sides of the projection (using a 512x512 pixel camera, for example), and would want benchmarks for the *square* whole-image. Therefore, I suggest that we modify benchmark.py to pass a rectangular half-image to each transform method. 

Specifically, I believe that modifying line 107 of abel/benchmarks.py from:
```python
x = np.random.randn(ni, ni)
```
to:
```python
x = np.random.randn(ni, np.floor(ni * 0.5) + 1)
```
should do the trick.

This has the added advantage of making all of the benchmarks seem a factor-of-two faster :). But, I think that most previous benchmarks use this convention. For example, when [Rallis et al](http://krex.k-state.edu/dspace/bitstream/handle/2097/18907/BenItzhakRevSciInst2014.pdf;sequence=1) say that they can transform a 1024x1024 image in 0.6 seconds, they are referring to the whole image. In PyAbel, we would be passing a 1024x512 half-image to the individual transform method. 

One could argue that, for the photoelectron distributions in Rallis et al, up/down symmetry could additionally be applied, and the transform could be performed on a 512x512 pixel image. However, this symmetry is not generic to the Abel transform, which only assumes cylindrical symmetry. Additionally, one could argue that, since the default behavior of `abel.Transform` is to allow for asymmetric transforms, where each quadrant is transformed separately, then the result of `abel.benchmark.AbelTiming` with n=101, actually corresponds to 25% of the time it would take to perform the transform on a 201x201 pixel image (which is not necessarily equal to the time it takes for a 101x101 image). 

Thoughts?

**Comment by stggh** [comment]
You are opening a can of worms:
1)  `linbasex` does require a square image, however, it is a special case in terms of the methods. 
2) I believe the basis methods form a whole image for processing a single quadrant in any case(*)

In general symmetry considerations improve the execution time by the symmetry divisor. Here are some `3pt` timings for a `8192x8192` VMI (don't worry about the even size "camera" image)
```python
In [1]: import numpy as np

In [2]: import abel

In [3]: IM = np.loadtxt("O2-ANU8192.txt")

In [4]: Q = abel.tools.symmetry.get_image_quadrants(IM)

In [5]: %timeit abel.Transform(IM, method='three_point')
4.77 s ± 57.9 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

In [6]: %timeit abel.dasch.three_point_transform(Q[0])
962 ms ± 5.98 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

In [7]: abel.benchmark.AbelTiming(n=[4096], select=['three_point'])
Out[7]: 
PyAbel benchmark run on x86_64

time in milliseconds

=========    Basis Abel implementations ==========

Implementation      n = 4096      
----------------------------------------
  three_point_bs    3046.3         

=========  Inverse Abel implementations ==========

Implementation      n = 4096      
----------------------------------------
     three_point     877.2         

```
(*) `basex`:
```python
In [1]: import numpy as np

In [2]: import abel

In [3]: IM = np.loadtxt("O2-ANU1024.txt.bz2")   # NB smaller image, I do not have all day ...

In [5]: %timeit abel.Transform(IM, method='basex')
345 ms ± 858 µs per loop (mean ± std. dev. of 7 runs, 1 loop each)   # forms one whole image each time?

In [6]: Q = abel.tools.symmetry.get_image_quadrants(IM)

In [7]: %timeit abel.basex.basex_transform(Q[0])

93.8 ms ± 85.1 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)
```

If we are planning to change benchmark, we should change `abel.Transform()` to process 1/2-images. `transform_image_by_half()`?

This may be worth investigating ...

**Comment by DanHickstein** [comment]
I don't think that we need to change `abel.Transform` at this point. I think that the main question here is: "What kind of Abel transform benchmark is the most informative?" In other words, how are most people using the numerical inverse Abel transform? What kind of symmetry are they assuming?

We can't know all the use-cases, but most people will be collecting images using digital cameras (which typically collect images that are square, or almost square), and they will be collecting both sides of the object. By definition of the Abel transform, the object must be left/right symmetric. abel.Transform does offer the "advanced" option to Abel transform the left and right sides of the image separately, but most people will want to assume cylindrical symmetry. This is what the old BASEX.exe does.

So, for bechmarking, I think that most people will want an answer to the question: "If I collect a 1024x1024 image using my digital camera, how long is it going to take to complete the left-right-symmetric Abel transform?"

We can answer this question by transforming a 1024x512 pixel half-image, and referring to this as a 1024x1024 image (okay, sizes must be odd, but just trying to give the general idea). This is sort of "cheating", but I think that most of the previous benchmarks in the literature have done something similar.

So, for the benchmarks to make sense, we need to make some assumption about the shape and symmetry of the images that users will be transforming. We could assume whatever we want, but I think that it makes the most sense to assume a square image with left/right symmetry applied.

BASEX and LinBASEX shouldn't really be disadvantaged by this approach. BASEX already assumes left/right symmetry. For some reason, it generates the basis set for the entire 1024x1024 image, instead of symmetrizing the image and simplifying the problem. Likely the calculation efficiency could be improved if the symmetry was applied to simplify the problem. Linbasex should have an advantage, since it assumes top/bottom symmetry (right?) and could reduce the complexity of the problem.

I'll go ahead and make a PR for the small change to the code in benchmarks.py (unless there are additional insights).

**Comment by DanHickstein** [comment]
This was addressed by #209, so I'm closing this issue.
